src/components/Sidebar/ActionNode.tsx

```diff
@@ -1,49 +1,212 @@
+        action.resource.resourceType = actionType?.label;
+      } else {
+        action = {
+          type: 'create',
+          description: '',
+          resource: {
+            id: shortid.generate(),
+            resourceType: actionType?.label,
+            code: {
+              coding: [
+                {
+                  system: '',
+                  code: '',
+                  display: ''
+                }
+              ]
+            }
+          }
+        };
+      }
+
+      updatePathway(setStateAction(pathway, currentNodeKey, [action]));
+    },
+    [currentNodeKey, pathway, updatePathway]
+  );
+
+  const selectCodeSystem = useCallback(
+    (event: ChangeEvent<{ value: string }>): void => {
+      if (!currentNodeKey) return;
+
+      const codeSystem = event?.target.value || '';
+      const currentState: GuidanceState = pathway.states[currentNodeKey] as GuidanceState;
+      const action: Action = currentState.action[0];
+      action.resource.code.coding[0].system = codeSystem;
+      updatePathway(setStateAction(pathway, currentNodeKey, [action]));
+    },
+    [currentNodeKey, pathway, updatePathway]
+  );
   // The node has a key and is not the start node
   const changeNodeTypeEnabled = currentNode.key !== undefined && currentNode.key !== 'Start';
 
+  const action = currentNode.action;
+  const resource = action && action.length > 0 && action[0].resource;
   // If the node does not have transitions it can be added to
   const displayAddButtons = currentNode.key !== undefined && currentNode.transitions.length === 0;
-
   return (
     <>
       {changeNodeTypeEnabled && (
-        <DropDown
-          id="nodeType"
-          label="Node Type"
-          options={nodeTypeOptions}
-          onChange={selectNodeType}
-          value="action"
-        />
+        <>
+          <DropDown
+            id="nodeType"
+            label="Node Type"
+            options={nodeTypeOptions}
+            onChange={selectNodeType}
+            value="action"
+          />
+          <DropDown
+            id="actionType"
+            label="Action Type"
+            options={actionTypeOptions}
+            onChange={selectActionType}
+            value={resource && resource.resourceType}
+          />
+          {(resource.resourceType === 'Medication Request' ||
+            resource.resourceType === 'Service Request') && (
+            <>
+              <DropDown
+                id="codeSystem"
+                label="Code System"
+                options={codeSystemOptions}
+                onChange={selectCodeSystem}
+                value={resource.code.coding[0].system}
+              />
+              {resource.code.coding[0].system && (
+                <Input
+                  id="code"
+                  label="Code"
+                  onChange={changeCode}
+                  value={resource.code.coding[0].code}
+                />
+              )}
+              {resource.code.coding[0].system && (
```

Comment:
If this was intended to be hidden until a system is selected then is there a reason it can't be included with the `Input` above it? Perhaps it should be hidden until a code is set though